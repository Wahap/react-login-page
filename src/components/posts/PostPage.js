import PropTypes from "prop-types";
import { connect } from "react-redux";
import React from "react";
import * as postActions from "../../redux/actions/postActions";
import * as apiActions from "../../redux/actions/apiStatusActions";
import { bindActionCreators } from "redux";
import Spinner from "../common/Spinner.js";
class PostPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    const { actions } = this.props;

    actions.getPosts().catch(error => {
      alert("Loading posts failed" + error);
    });
  }

  render() {
    return (
      <>
        {this.props.loading ? (
          <>
            <Spinner />
            <div>
              {" "}
              <button
                style={{ marginBottom: 20 }}
                className="btn btn-primary add-course"
                onClick={() => this.stopApiCall()}
              >
                stop
              </button>
              <button
                style={{ marginBottom: 20 }}
                className="btn btn-primary add-course"
                onClick={() => this.startApiCall()}
              >
                arttirt
              </button>
            </div>
          </>
        ) : (
          <div>
            <p>{this.props.wahap}++</p>
            {this.props.posts.map(user => (
              <p>{user.email}</p>
            ))}
            test
            <button
              style={{ marginBottom: 20 }}
              className="btn btn-primary add-course"
              onClick={() => this.startApiCall()}
            >
              Add Course
            </button>
          </div>
        )}

        <div>
          <p>{this.props.wahap}++</p>

          <button
            style={{ marginBottom: 20 }}
            className="btn btn-primary add-course"
            onClick={() => this.startApiCall()}
          >
            Add Course
          </button>
        </div>
      </>
    );
  }
}

PostPage.propTypes = {
  posts: PropTypes.array.isRequired,
  wahap: PropTypes.bool.isRequired
};

function mapStateToProps(state) {
  return {
    posts: state.posts,
    loading: state.apiCallsInProgress > 0,
    wahap: "test" + state.apiCallsInProgress + 2
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: {
      getPosts: bindActionCreators(postActions.getPosts, dispatch),
      beginApiCall: bindActionCreators(apiActions.beginApiCall, dispatch),
      stopApiCall: bindActionCreators(apiActions.stopApiCall, dispatch)
    }
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PostPage);
