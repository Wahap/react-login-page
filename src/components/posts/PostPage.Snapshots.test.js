import React from "react";
import renderer from "react-test-renderer";
import { courses, authors } from "../../../tools/mockData";
import PostPage from "./PostPage";
import configureStore from "../../redux/configureStore";
const store = configureStore();

const props = {
  loading: true
};

it("sets submit button label 'Saving...' when saving is true", () => {
  const tree = renderer.create(<PostPage store={store} {...props} />);

  expect(tree).toMatchSnapshot();
});
