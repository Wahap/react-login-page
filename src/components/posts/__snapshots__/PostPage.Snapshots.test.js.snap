// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit button label 'Saving...' when saving is true 1`] = `
<div
  className="loader"
>
  Loading...
</div>
`;
